File: python/7 kyu/430.py
# https://www.codewars.com/kata/5a731b36e19d14400f000c19/train/python

# Passed

def decode_pass(pass_list, bits):
    password_from_binary = "".join(chr(int(bit, 2)) for bit in bits.split(' '))
    result = password_from_binary if password_from_binary in pass_list else False
    return result

output = decode_pass(['password123', 'admin', 'admin1'], '01110000 01100001 01110011 01110011 01110111 01101111 01110010 01100100 00110001 00110010 00110011')
print(output)

File: nodejs/6 kyu/430.js
// https://www.codewars.com/kata/54f9f4d7c41722304e000bbb/train/javascript

// Passed

function firstDup(string) {
  let charToFirstIdxMap = new Object();
  let charToCountMap = new Object();

  for (let idx = 0; idx < string.length; idx++) {
    let char = string[idx];
    if (!(char in charToFirstIdxMap)) {
      charToFirstIdxMap[char] = idx;
    }

    if (!(char in charToCountMap)) {
      charToCountMap[char] = 0;
    }

    charToCountMap[char]++;
  }

  let result;
  for (const char of Object.keys(charToCountMap)) {
    let charCount = charToCountMap[char];
    if (charCount === 1) {
      continue;
    }

    if (
      result === undefined ||
      charToFirstIdxMap[char] < charToFirstIdxMap[result]
    ) {
      result = char;
    }
  }

  return result;
}

const output = firstDup("tweet");
console.log(output);